pkg/nfd-topology-updater/nfd-topology-updater.go

```diff
@@ -199,6 +200,7 @@ func (w *nfdTopologyUpdater) updateNodeResourceTopology(zoneInfo v1alpha2.ZoneLi
 
 	nrtMutated := nrt.DeepCopy()
 	nrtMutated.Zones = zoneInfo
+	nrtMutated.Attributes = append(nrt.Attributes, scanResponse.Attributes...)
```

Comment:
Shouldn't we check existing attributes and replace old attributes with the same name instead of blindly appending?

Reply:
Good point ...  I have just created a new function although it could be a better way to merge them..

Reply:
> I happened to try to run the e2e tests but failed on two different clusters with
> 
> ```
>     Feb 15 11:52:06.023: INFO: podFingerprint attributte "nodeTopologyPodsFingerprint" not found attribute "nodeTopologyPodsFingerprint" not found
>     Feb 15 11:52:15.519: FAIL: Timed out after 60.003s.
> ```

it seems like the Attribute is not being created in the NRT object ... 
I have checked and nothing pretty obvious pop up, so I need to look a little deeper. 
I will take a look next monday

Reply:
I have changed the moment when podfingerprint is computed. 
I was doing if AFTER the resources were filtered (in podresourcescanner.go:152) and if no resource pass the filter the Attribute was not added at all. 
Now it is computed using all the resources which seems better because podfingerprint should resemble node resource status